### ISSUE TITLE
Error in Style/EvalWithLocation

### ISSUE THREAD

**Issue by 6temes** [rubocop-hq/rubocop#9477]
This code makes the cop Style/EvalWithLocation crash:

```ruby
SimpleForm::Inputs::Base.class_eval do
  # We just need the ability to pass a block on the `use` function
  # then we save it on the Leaf class to be able to call it
  def merge_wrapper_options options, wrapper_options
    if wrapper_options
      wrapper_options.merge options do |key, oldval, newval|
        case key.to_s
        when 'class'
          classes = Array(oldval) + Array(newval)
          classes << 'is-invalid' if has_errors?
          classes
        when 'data', 'aria'
          oldval.merge newval
        else
          newval
        end
      end
    else
      options
    end
  end
end
```

Rubocop 1.9.0 (using Parser 3.0.0.0, rubocop-ast 1.4.1, running on ruby 2.7.2 x86_64-darwin20)

**Comment by dvandersluis** [comment]
Please see the pinned issue. Already fixed by #9435.

**Comment by 6temes** [comment]
Upps. Sorry. :)
